FreeRTOS-Plus/Source/Application-Protocols/network_transport/freertos_plus_tcp/using_plaintext/using_plaintext.c

```diff
@@ -122,14 +122,24 @@ int32_t Plaintext_FreeRTOS_recv( NetworkContext_t * pNetworkContext,
 {
     PlaintextTransportParams_t * pPlaintextTransportParams = NULL;
     int32_t socketStatus = 0;
+    BaseType_t recvCount = 0;
 
     configASSERT( ( pNetworkContext != NULL ) && ( pNetworkContext->pParams != NULL ) );
 
     pPlaintextTransportParams = pNetworkContext->pParams;
-    socketStatus = FreeRTOS_recv( pPlaintextTransportParams->tcpSocket,
-                                  pBuffer,
-                                  bytesToRecv,
-                                  0 );
+
+    if( ( bytesToRecv > 1 ) ||
```

Comment:
Is this supposed to be >= to 1?

Reply:
I think the intention here is that if `bytesToRecv` is 1, it is counted as a speculative read, so we want to check if there are bytes in the stream that are yet to be read. Otherwise, we just want to block for the entire duration of the timeout.

Reply:
Ok, so when coreMQTT goes to read a single byte for the packet type, if there is data FreeRTOS_recvcount will be greater than zero.
But if there is no data, then zero will be returned. In this case, do we not still want to try to read from the socket for the socket's recv timeout?

Reply:
It deviates from the original behavior, but I believe Richard made this suggestion so that the receive block time doesn't put the task to sleep when an attempt is made to read a single byte.

Reply:
Ok, so I guess we shall see if there are more nodataavailable failures in the demo after this change.

Reply:
Sounds good. I ran a custom job and looks like everything still passed in CI.

Reply:
I'm concerned with the special logic of the _speculative read_. Why make a special case only for a single byte? There is no requirement of this from the `transport_interface.h` specification. 
We don't have this behavior for the POSIX implementation or the Secure Sockets based implementation. So a special case just for FreeRTOS+TCP implementation does not sound good.

Reply:
This is suggested by @RichardBarry in an earlier comment.

Reply:
Can you please add a comment in the code explaining the rationale? 
Also, it would be good to mention this in the API documentation for the function about the single read not waiting for the socket timeout period so that the calling code can be aware of utilizing this feature of the implementation

Reply:
I updated the comment in the code.